apistar/compat.py

```diff
@@ -0,0 +1,32 @@
+import collections
+import sys
+
+if sys.version_info < (3, 6):
+    dict_type = collections.OrderedDict
+else:
+    dict_type = dict
```

Comment:
Not sure, that we can assume, that dict has ordered keys since 3.6

Reply:
dict is ordered in cpython 3.6 as an implementation detail, and guaranteed ordered in 3.7+

We don't *strictly* need ordering, it's only for stylistic purposes of eg. returning JSON outputs that are always in the same way as the fields in the codebase, so it's not a massive problem if there are a few edge cases of platforms that *could* get unordered outputs, but I'd be happy if there's any stricter checking that we can do, to also check if we're on `cpython` or some other implementation.

Reply:
> The order-preserving aspect of this new implementation is considered an implementation detail and should not be relied upon

https://docs.python.org/3/whatsnew/3.6.html#whatsnew36-compactdict

I could not find anything about guaranteed dict order since 3.7+.

Reply:
The boss said it : https://mail.python.org/pipermail/python-dev/2017-December/151283.html